### ISSUE TITLE
Reproduce results on cifar10

### ISSUE THREAD

**Issue by MrChenFeng** [google-research/simclr#60]
#22 
I have tried resnet18(batch size 1024) and also resnet50(batch size 512) as the backbone, learning rate 1.2 with cosineannealingLR:
the linear classifier is another three-layer perceptron. However, I can only achieve ~89.5 in cifar10. Before I got 93% accuracy but soon I found I'm actually fine-tuning the whole network(forgot to froze the weights of resnet).
For resnet50, I applied following hyperparameters,
![image](https://user-images.githubusercontent.com/24207392/87832623-52def780-c87e-11ea-980e-e89a0f772d7d.png)
The training loss:
![image](https://user-images.githubusercontent.com/24207392/87832854-d567b700-c87e-11ea-882b-23fe9815acaf.png)
The augmentations for training:
![image](https://user-images.githubusercontent.com/24207392/87832911-fdefb100-c87e-11ea-97db-7b2f64bc1f34.png)
The augmentations for evaluation:
![image](https://user-images.githubusercontent.com/24207392/87832948-1069ea80-c87f-11ea-8fa9-98387c0197aa.png)

Highly appreciated for suggestions from anyone who has succeeded in reproduce the cifar10.

**Comment by MrChenFeng** [comment]
I actually used cosine_annelaing for this SGD run and the loss seems to be converged in my run. I used Adam with 1e-3 also but the performance is even worse. In summary, I tried different batch_size(256,512,1024)/projection_dim(64,128,512)/Optimizer(SGD, Adam)/Epochs(100,500,1000)/network(resnet18,resnet50), I still can't reproduce the result. 
If possible, could you provide a pretrained network for cifar10, thanks so much.
And also, can I confirm something with you here:
1. Do you use any other image transformations besides ToTensor() in the linear evaluation stage?
2. The performance you reported is pure linear evaluation rather than fine-tune the whole network right?
3. Do you upscale the cifar10 from 32x32 to 224x224 by interpolation? In the paper, you said you changed the network rather than to change the input size.
4. Have you tried resnet18 also? Basically I used resnet18 as backbone cause it's faster, however I tried resnet50 also but the performance is similar to resnet18, I still can't reproduce.

**Comment by chentingpc** [comment]
Please see readme for instructions on how to run cifar10 using this TF repo. You're re-implementing it using pytorch, so there may be many places that things could go wrong (for which you need to refer to other successful re-implementation of simclr in pytorch as linked in the repo).

To answer your questions:
1. we only use random cropping (the same as ImageNet, which may not be necessary for cifar-10 but haven't tested) during linear eval.
2. no fine-tuning for cifar10 reported.
3. we use 32x32
4. in the readme, there's command for obtaining resnet18 result by running the current repo.

**Comment by noah822** [comment]
Hi MrChenFeng! I am recently working on reproducing this paper on cifar10 out of interest. My experiment layout is as follows
1. I do $batchsize = 512$ and $epoch=100$, and basically same augmentation scheme as yours
2. use ResNet50 as backbone, followed by MLP with relu activation $2048 \rightarrow 2048 \rightarrow 128$
3. I use LARS optimizer with learning rate $0.075 \times \sqrt{batch size} \approx 1.5$ with 10 epoch linear warm up and cosine annealed to 1e-3 in 90 epoch. 

I can only achieve round 5.4 average batch loss, while yours is around 5.0. And my linear evaluation loss is extremely poor, where I attach a $Linear(2048, 10)$ head with Nesterov momentum optimizer with $lr=0.1 \times batch size /256 \approx 0.2$. I can not even converge with this setting. Previously, I have tried Adam with $1e-3$ learning rate, where I can only achieve 60% something accuracy. There is a huge gap between my result and that reported in the paper.

I know it has been a long time since the simclr paper comes out. I would appreciate if you can give some hints in terms of linear evaluation setting and something probably incorrect in simclr pretraining. I've tried multiple times, but still cannot even make it. I thought there might be some severe bug with my implementation.
